modules/ROOT/pages/creating-connected-apps-dev.adoc

```diff
@@ -0,0 +1,195 @@
+The access token can now be used to call Anypoint Platform APIs on the user’s behalf.
+
+[[refresh-token]]
+=== Refresh Token
+
+Use the `refresh_token` grant type to grant access to user data when the user isn’t signed in. Refresh tokens expire after 90 days. Access tokens expire after the user’s session expires.
+
+To work with OpenID Connect, apps have to exchange refresh tokens for new access tokens after the initial authorization to continue accessing user data.
+
+If the app is using the authorization code grant type, the `offline_access` scope is required to use refresh tokens after the original access tokens expire.
+
+To request a new access token, send a request to the token endpoint using the `refresh_token` grant type. When authentication is successful, a refresh token is exchanged for an access token.
+
+[[password]]
+=== Password
+
+Use the `password` grant type when the application exchanges the user’s username and password for an access token.
+
+[[NOTE]]
+This grant type is not recommended, as it exposes user credentials directly to the connected app. It also precludes the use of additional security measures such as multi-factor authentication.
+
+[[jwt-bearer]]
+=== JWT Bearer Token
+
+Use the `jwt_bearer` (`urn:ietf:params:oauth:grant-type:jwt-bearer`) grant type when the application receives access tokens without transmitting sensitive information, such as the client secret.
+
+The `jwt_bearer` grant type is suitable for use with trusted clients to gain access to protected resources without user authorization, for example:
+
+. A developer creates the connected app that acts on behalf of a user with `jwt_bearer` as the grant type.
+. The JWT token exchange occurs with these claims in the payload:
++
+|===
+| Attribute | Description
+
+| `iss`
+| The issuer of the claim. This value must be same as the connected app’s `clientId`.
+
+| `sub`
+a| The subject of the token, such as:
+----
+{version}\|{identityProviderId}\|{username}\|{firstname}\|{lastname}\|email\|externalgroups
+----
+
+NOTE: At minimum, `identityProvider_id` and `username` are required for `sub`, as these subjects identify the user in the system.
+
+| `aud`
+| The authentication server. For example, the authentication server for the US region is `https://anypoint.mulesoft.com/accounts/api/v2/oauth2/token`.
+
+|===
+. The app calls the authorization server with the generated assertion.
+
+[[client-credentials]]
+=== Client Credentials
+
+Use the `client_credentials` grant type for machine-to-machine authentication where a specific user’s permission to access data isn't required.
+
+The `client_credentials` grant type uses the `clientId` and `clientSecret` of the client to authorize and access protected data.
+
+You can configure CI/CD pipelines using service accounts to programmatically call Anypoint Platform to use connected apps to prevent service disruptions when configuring xref:multi-factor-authentication.adoc[multi-factor authentication].
```

Comment:
Maybe put user value (prevent service disruptions) first?

In any case, this sentence is a long drink of water...